jsonsl.c

```diff
@@ -424,6 +424,21 @@ jsonsl_feed(jsonsl_t jsn, const jsonsl_char_t *bytes, size_t nbytes)
                     VERIFY_SPECIAL("false");
                 } else if (state->special_flags == JSONSL_SPECIALf_NULL) {
                     VERIFY_SPECIAL("null");
+                } else if (state->special_flags == JSONSL_SPECIALf_NAN) {
+                   /* like VERIFY_SPECIAL but case-insensitive */
+                   if (tolower(CUR_CHAR) != "nan"[jsn->pos - state->pos_begin]) {
+                      INVOKE_ERROR (SPECIAL_EXPECTED);
+                   }
+                } else if (state->special_flags & JSONSL_SPECIALf_NULL ||
+                           state->special_flags & JSONSL_SPECIALf_NAN) {
+                   /* previous char was "n", are we parsing null or nan? */
+                   if (CUR_CHAR != 'u') {
+                      state->special_flags &= ~JSONSL_SPECIALf_NULL;
+                   }
+
+                   if (tolower(CUR_CHAR) != 'a') {
```

Comment:
Actually it's two "!=" statements, so CUR_CHAR might be neither 'a', 'N', nor 'n', I think we have to check both if statements.